src/bot/handler.rs

```diff
@@ -42,14 +67,28 @@ impl Handler {
         log::info!("Starting the El Monitorro bot");
 
         let interval = std::time::Duration::from_secs(1);
-
         loop {
             while let Some(update) = api.next_update() {
                 let db_pool = crate::db::pool().clone();
                 let tg_api = api.clone();
 
-                thread_pool
-                    .spawn(move || Self::process_message_or_channel_post(db_pool, tg_api, update));
+                match update.content.clone() {
```

Comment:
I think here you can not `clone` not sure, I am sure you don't need message or callback refs you can just match with variant and inside parentheses put this pattern `_`